Makefile

```diff
@@ -20,5 +21,22 @@ lint: install-golangci-lint
 	@echo "> Linting code..."
 	@golangci-lint run -c golangci.yaml
 
-test: format lint
+tools/bcdhive_generator/.build: $(wildcard tools/bcdhive_generator/*)
+ifneq ($(OS),Windows_NT)
+	@echo "> Building bcdhive-generator in Docker using current golang version"
+	docker build tools/bcdhive_generator --tag bcdhive-generator --build-arg go_version=$(GO_VERSION)
+
+	@touch tools/bcdhive_generator/.build
+else
+	@echo "> Cannot generate on Windows"
+endif
+
+layer/bcdhive_generated.go: layer/windows_baselayer.go tools/bcdhive_generator/.build
+ifneq ($(OS),Windows_NT)
+	$(GOCMD) generate ./...
+else
+	@echo "> Cannot generate on Windows"
+endif
```

Comment:
@ekcasey I changed targets to be the actual files, with the `docker build` target generating a gitignored `.build` file that is timestamp-checked along with the rest of the contents.

This still feels a little busy... I wonder if putting all this into either sub-Makefiles would be preferrable. Otherwise, we could also just fold all this into a single target.

Reply:
lgtm for now. feel free to submit a cleanup PR that combines them if you think it looks better